doc/ui-guidelines.md

```diff
@@ -49,36 +57,46 @@ NOW:
 - State values no longer need to be dereferenced; they are accessible as regular symbols. This eliminates a common bug where the "@" symbol was inadvertently omitted.
 - `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
 - `:f>` not needed anymore, all components are functional by default
-- `rn/use-callback` hook should be used for anon callback functions
 
 > [!IMPORTANT]
-> DO NOT USE anon functions directly in the props
- 
+> For components that re-render frequently, ensure you pass stable reference as props, this might have you wrapping your callback in `rn/use-callback` or your maps/vectors in `rn/use-memo`
+
+An example of such case would be when we have a heavy chat component which we only want to re-render when the props truly change, we can memoize callbacks/anon function passed to it.
+
 BAD
 ```clojure
 (defn view
-  []
-  (let [[pressed? set-pressed] (rn/use-state false)]
-    [rn/pressable
-     {:style               (style/main pressed?)
-      :on-press-in         #(set-pressed true)
-      :on-press-out        #(set-pressed nil)}]))
+  [chat-id]
+  (let [[opened? set-opened] (rn/use-state false)]
+    [heavy-chat-component
+     {:style                (style/main opened?)
+      :chats                (get-chats chat-id)
+      :on-chat-open         #(set-opened true)
+      :on-chat-close        #(set-opened nil)}]))
 ```
 
 GOOD:
 ```clojure
 (defn view
   []
-  (let [[pressed? set-pressed] (rn/use-state false)
-        on-press-in            (rn/use-callback #(set-pressed true))
-        on-press-out           (rn/use-callback #(set-pressed nil))]
-    [rn/pressable
-     {:style               (style/main pressed?)
-      :on-press-in         on-press-in
-      :on-press-out        on-press-out}]))
+  (let [[opened? set-opened]  (rn/use-state false)
+        chats                 (rn/use-memo #(get-chats chat-id)) [chat-id]
+        on-chat-open          (rn/use-callback #(set-pressed true))
+        on-chat-close         (rn/use-callback #(set-pressed nil))]
+    [heavy-chat-component
+     {:style                (style/main opened?)
+      :chats                chats
+      :on-chat-open         on-chat-open  
+      :on-chat-close        on-chat-close}]))
 ```
 
-## Global state and subscriptions
+Note that this should be done for only components that re-renders more frequently, as so you should profile before using. This is because you might not bebefi from using it much in scenarios 
```

Comment:
```suggestion
Note that this should be done for only components that re-renders more frequently, as so you should profile before using. This is because you might not benefit from using it much in scenarios 
```